packages/template-retail-react-app/app/ssr.js

```diff
@@ -36,65 +35,10 @@ const options = {
 const runtime = getRuntime()
 
 const {handler} = runtime.createHandler(options, (app) => {
-    const getRuntimeEnv = () => {
-        if (process.env.NODE_ENV !== 'production') return process.env.NODE_ENV ?? 'development'
-        const origin = getAppOrigin()
-        // mobify-storefront-staging sites have NODE_ENV set to production, but for the purposes
-        // of CSP we consider the sites to be staging.
-        return origin.endsWith('.mobify-storefront-staging.com') ? 'staging' : 'production'
-    }
-
-    // This is a temporary solution while we work on Storefront Preview - a full solution will
-    // prevent required CSP headers from being modified.
-    const getCSP = () => {
-        const trustedMap = {
-            development: [
-                'localhost:*',
-                '*.commercecloud.salesforce.com',
-                '*.demandware.net',
-                '*.mobify-staging.com',
-                '*.mobify-storefront-staging.com',
-                '*.mobify-storefront.com',
-                'runtime.commercecloud.com'
-            ],
-            staging: [
-                '*.demandware.net',
-                '*.mobify-staging.com',
-                '*.mobify-storefront-staging.com',
-                '*.mobify-storefront.com',
-                '*.commercecloud.salesforce.com',
-                'runtime.commercecloud.com'
-            ],
-            production: [
-                '*.demandware.com',
-                '*.mobify.com',
-                '*.mobify-storefront.com',
-                '*.commercecloud.salesforce.com',
-                'runtime.commercecloud.com'
-            ]
-        }
-
-        const env = getRuntimeEnv()
-        const trusted = ["'self'", ...(trustedMap[env] ? trustedMap[env] : [])]
-
-        return {
-            'connect-src': ['api.cquotient.com', ...trusted],
-            'frame-ancestors': [...trusted],
-            'img-src': ['data:', ...trusted],
-            'script-src': ["'unsafe-eval'", 'storage.googleapis.com', ...trusted],
-
-            // Do not upgrade insecure requests for local development
-            'upgrade-insecure-requests': isRemote() ? [] : null
-        }
-    }
-
     // Set HTTP security headers
     app.use(
         helmet({
-            contentSecurityPolicy: {
-                useDefaults: true,
-                directives: getCSP()
-            },
+            contentSecurityPolicy: getContentSecurityPolicy(),
```

Comment:
We want to prevent customer from shooting their own foot, and it is currently possible by overriding `ssr.js` in their project.

We can make it **impossible** to override the http response headers by adding an Express middleware **after** the Express server instance is instantiated by `runtime.createHandler()`...

Reply:
Take a look at the ssr server implementation [here](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/525f88bb51ed12e590a7795bd85bb2d08801db31/packages/pwa-kit-runtime/src/ssr/server/build-remote-server.js#L922).

Or we can go even deeper to patch the `http` outgoing traffic, [here](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/525f88bb51ed12e590a7795bd85bb2d08801db31/packages/pwa-kit-runtime/src/ssr/server/build-remote-server.js#L1047)